Source/bmalloc/libpas/src/libpas/pas_report_crash.c

```diff
@@ -114,64 +114,98 @@ kern_return_t pas_report_crash_extract_pgm_failure(vm_address_t fault_address, m
             report->alignment = "address right-aligned";
 
             // Right-aligned "Lower PGM OOB" checking
-            // FIXME: This assumes the allocation is still live.
             addr64_t bottom = (addr64_t) (key - pgm_metadata->mem_to_waste - pgm_metadata->page_size);
             addr64_t top = (addr64_t) (key - pgm_metadata->mem_to_waste);
 
-            if (pas_fault_address_is_in_bounds(fault_address, bottom, top))
-                return pas_update_report_crash_fields(report, "long-range OOB", "low", fault_address, pgm_metadata->allocation_size_requested);
+            if (pas_fault_address_is_in_bounds(fault_address, bottom, top)) {
+                if (pgm_metadata->free_status) {
+                    // Physical addresses are released
+                    return pas_update_report_crash_fields(report, "long-range UAF", "low", fault_address, pgm_metadata->allocation_size_requested);
```

Comment:
You could probably do something like

```
return pas_update_report_crash_fields(report, pgm_metadata->free_status ? "long-range UAF" : "long-range OOB", "low", fault_address, pgm_metadata->allocation_size_requested);
```

This can apply to the cases below too.

Reply:
Fixed